screens/signUp.js

```diff
@@ -23,22 +28,56 @@ const SignUp = () => {
             style={tailwind("mb-4")}
             placeholder="name"
             icon={require("../assets/icons/black/person.png")}
+            onChange={(e) => setName(e)}
           />
           <IconInput
             style={tailwind("mb-4")}
             placeholder="email"
             icon={require("../assets/icons/black/at.png")}
             keyboardType="email-address"
+            onChange={(e) => setEmail(e)}
           />
           <IconInput
             style={tailwind("mb-4")}
             placeholder="password"
             icon={require("../assets/icons/black/key.png")}
             secureTextEntry={true}
+            onChange={(e) => setPassword(e)}
           />
         </View>
         <View>
-          <RoundButton style={tailwind("bg-green-500 mb-2")} title="sign up" />
+          <TouchableOpacity
+            onPress={() => {
+              const options = {
+                method: "POST",
+                headers: {
+                  "Content-Type": "application/json",
+                },
+                body: JSON.stringify({
+                  data: {
+                    name: name,
+                    email: email,
+                    password: password,
+                  },
+                }),
+              };
+
+              fetch("http://localhost:8000/api/db/addcred", options)
+                .then((res) => res.json())
+                .then((data) => setDbInsert(data.message))
```

Comment:
Unless you want to use this data somewhere, there is no need to set a state for it.